api/ReadableStream.json

```
@@ -92,6 +92,46 @@
           }
         }
       },
+      "async_iterable": {
+        "__compat": {
+          "description": "Async iterable (supports <code>for await (const chunk of readableStream) {}</code>.",
```

FYI @queengooborg @Elchi3 etc. This feature is just like `secure_context_required` in that it is a property of the API rather than an explicit interface. It is likely that other APIs may well share this feature in future so it may be worth spending a little time thinking about a standard key and some standard description text?

1. Key, I chose "async_iterable". I think it is pretty clear, but if you think we should have `supports_async_iterable` or `async_iterable_supported` let me know.
2. Is description too long? I mean you could have any of the following:
   - Async iterable
   - Supports async iteration
   - Supports async iteration using `await for`
   - and so on.
   
   I quite like having this because it immediately gives a flavour for what you can do. But I'm aware BCD likes to keep descriptions succint. Also you might want to enforce all instances of this having the same description, in which case something like:
   
   > "Async iterable (supports `for await (const item of iterable) {}` 

Thoughts?

---

Thoughts to get to a data guideline are in this issue: https://github.com/mdn/browser-compat-data/issues/6367
I made a proposal there a while ago for Iterable. Not sure if it still has consensus. If it does, the description for async iterable would probably be:

`"Async iterable (<code>@@asyncIterator</code>)"`

---

@Elchi3 Thanks. I'll modify it as you have suggested. Feel free to merge.
Note, 
- I followed the discussion and it wasn't clear to me from that what the consensus was.
- Note that my instincts and @foolip first ones were the same - to include the "await-for-of" in the description. These symbols are IMO largely "spec language" - even if you have seen them before you'll probably not think "Oh yeah, that means I can use `await for thingy of iterable`". That seems like a missed opportunity.

```suggestion
          "description": "Async iterable (<code>@@asyncIterator</code>)</code>",
```

---

Yeah, I guess the description needs to be hashed out still. I think avoiding "spec language" is a good point.